src/tensorflow_backend_tf.cc

```diff
@@ -932,31 +945,35 @@ TRITONTF_ModelCreateFromSavedModel(
   std::unique_ptr<tensorflow::SavedModelBundle> bundle(
       new tensorflow::SavedModelBundle);
 
+  // If user does not specify a 'tag' in the config.pbtx, use 'serve' as default
   std::unordered_set<std::string> saved_model_tags;
-  saved_model_tags.insert(tensorflow::kSavedModelTagServe);
+  static const std::string TAG_TO_USE = 
+      graph_tag.empty() ? tensorflow::kSavedModelTagServe : graph_tag; 
+  saved_model_tags.insert(TAG_TO_USE);
 
   tensorflow::RunOptions run_options;
   RETURN_IF_TF_ERROR(tensorflow::LoadSavedModel(
       session_options, run_options, model_path, saved_model_tags,
       bundle.get()));
 
-  // Verify that the bundle has the "serve" tag
+  // Verify that the bundle has the correct tag
   bool found_serve_tag = false;
   for (const auto& tag : bundle->meta_graph_def.meta_info_def().tags()) {
-    if (tag == tensorflow::kSavedModelTagServe) {
+    if (tag == TAG_TO_USE) {
       found_serve_tag = true;
       break;
     }
   }
   if (!found_serve_tag) {
     return TRITONTF_ErrorNew(
         "unable to load model '" + std::string(model_name) + "', expected '" +
-        tensorflow::kSavedModelTagServe + "' tag");
+        TAG_TO_USE + "' tag");
   }
 
-  // Verify that a "serving_default" signature exists, that is what
-  // will be used to verify the inputs and outputs.
-  static const std::string DEFAULT_SERVING_SIGNATURE_DEF_KEY("serving_default");
+  // If user does not specify a 'signature_def' in the config.pbtx,
+  // then use "serving_default" as default
+  static const std::string DEFAULT_SERVING_SIGNATURE_DEF_KEY = 
```

Comment:
We should not change the DEFAULT_SERVING_SIGNATURE_DEF_KEY, Rather create a new variable SIGNATURE_DEF_KEY_TO_USE like you did for TAG to use since it is technically not the default signature key if user provides the value.